### PR TITLE
Improve error reporting

### PR DESCRIPTION
**Pull request by medikoo**
With goal to improve specificity of error codes in telemetry reports

- Bug fixes:
   - Ensure user errors are thrown with `ServerlessError` constructor (otherwise they're assumed as _programmer error_)
   - In local invocation report invalid (non function) handler meaningfully (bug noticed in telemetry reports)
   - Ensure to not dirty crash on misconfigured resources when normalizing resources for _check for changes_ check (user may inject incomplete resource configuration to `resources` and crashing here on that, prevented failing with meaningful error message)
- Telemetry enhancements
  - If we deal with user error, which has error code, but it's format is not familiar, report also error location (it's to pick eventual omissions in setting meaningful error code, where e.g. external error code is passed, in telemetry such case can be observed with `ValidationError` which was not perfectly clear where it comes from)
  - Introduce specific error codes for AWS request error. now error code will include information about service and method that failed
  - Introduce specific error codes for Stack update failures. Property validation is detected specifically, otherwise we report service, method name and error type within error code
- Maintenance improvements
   - Move local invocation test fixtures to test folder
   - Remove obsolete (dead path) error handling in implementation of `sls stats` command

### CODE REVIEW

File: lib/classes/PluginManager.js [hunk 1]
```diff
@@ -372,24 +372,31 @@ class PluginManager {
...
+          'CUSTOM_LEGACY_VARIABLES_RESOLVER_INVALID_FORMAT'
         );
       }
       if (options.isDisabledAtPrepopulation && typeof options.serviceName !== 'string') {
-        throw new Error(
-          `Custom variable resolver for ${variablePrefix} defined by ${pluginName} specifies isDisabledAtPrepopulation but doesn't provide a string for a name`
+        throw new ServerlessError(
+          `Custom variable resolver for ${variablePrefix} defined by ${pluginName} ` +
+            "specifies isDisabledAtPrepopulation but doesn't provide a string for a name",
+          'USTOM_LEGACY_VARIABLES_RESOLVER_INVALID_CONFIGURATION'
```
Comment (pgrzesik) [inline lib/classes/PluginManager.js:399]: There's `C` missing at the beginning of error code

File: lib/cli/handle-error.js [hunk 2]
```diff
@@ -167,7 +169,7 @@ module.exports = async (exception, options = {}) => {
       code: exception.code,
     };
 
-    if (!isUserError || !exception.code) {
+    if (!isUserError || !exception.code || !isErrorCodeNormative(exception.code)) {
```
Comment (pgrzesik) [inline lib/cli/handle-error.js:172]: Great call with including `location` for such errors as well

**Review by pgrzesik — APPROVED** [review]
Awesome job :clap: I've managed to spot only a minor typo in one of the error codes, otherwise it looks great

File: lib/classes/PluginManager.js (hunk 1 above)
Comment (medikoo) [reply lib/classes/PluginManager.js]: Good find! Fixed

**Review by pgrzesik — APPROVED** [review]
Well done :clap:

**Pull request merged**